src/Microsoft.Health.Fhir.Api/Features/Context/FhirRequestContextMiddleware.cs

```diff
@@ -41,6 +42,12 @@ public async Task Invoke(HttpContext context, IFhirRequestContextAccessor fhirRe
                 request.QueryString);
 
             string correlationId = correlationIdProvider.Invoke();
+            var key = context.Request.Headers.Keys.FirstOrDefault(h => h.Equals(KnownHeaders.RequestId, StringComparison.OrdinalIgnoreCase));
```

Comment:
Would `context.Request.Headers.Contains` or `context.Request.Headers.TryGetValues` be better here?

Reply:
Contains can accept `StringComparison` enum but you still end need to find right key (it can be "x-Request-Id" or "X-REQUEST-ID") and you don't know which one are you looking.
TryGetvalues doesn't accept `StringComparison` at all.


This one is not perfect if you get two headers we basically flip coin which one to choose, but i'm not sure it's worth validation...

---
In reply to: [589825635](https://github.com/microsoft/fhir-server/pull/1728#discussion_r589825635) [](ancestors = 589825635)